redirect-rules/simple/rule.js

```diff
@@ -0,0 +1,36 @@
+function redirectToConsentForm (user, context, callback) {
+  var hasConsented = user.app_metadata && user.app_metadata.has_consented;
+
+  // redirect to consent form if user has not yet consented
+  if (!hasConsented && context.protocol !== 'redirect-callback') {
+    var auth0Domain = auth0.baseUrl.match(/([^:]*:\/\/)?([^\/]+\.[^\/]+)/)[2];
+
+    context.redirect = {
+      url: configuration.CONSENT_FORM_URL +
+        '&auth0_domain=' + encodeURIComponent(auth0Domain) +
+        // generate random state string that Auth0 will validate when we return
+        '&state=' + encodeURIComponent(new Buffer(Math.random().toString()).toString('base64'))
```

Comment:
@tjanczuk: to my knowledge we do not validate the `state` parameter yet. But I've been advised to include it for forward compatibility. This code at least generates a somewhat random/unique hash that, when validation does exist, will protect the implementor from invalid calls to `/continue`. And FYI - I kept the implementation of the hash generation somewhat generic (didn't use an npm module like [node-uuid](https://github.com/broofa/node-uuid)) because I wanted something that would run on an Auth0 appliance as well as in the cloud.